docs/topics/native/native-c-interop-stability.md

```diff
@@ -0,0 +1,180 @@
+[//]: # (title: Stability of using C and Objective-C libraries)
+
+## Intro
+
+Kotlin/Native provides [the ability to use C](native-c-interop.md) and [Objective-C](native-objc-interop.md) libraries.
+This document calls them "native libraries" for simplicity.
+
+This feature is currently [in Beta](components-stability.md).
+One of the main reasons for this is that using native libraries might affect the compatibility of your code
+with different versions of Kotlin, dependencies and Xcode.
+In other words, the compatibility might be worse compared to the case when your code doesn't use native libraries.
+This document covers this topic.
+
+> This document describes not only problems that do happen often in practice, but also problems that might happen
+> (but usually don't), and hypothetical potential problems as well.
+>
+{style="note"}
+
+There is an important distinction:
+- On each platform, Kotlin provides access to "system" native libraries through so-called
+  ["platform libraries"](native-platform-libs.md)
+- All other native libraries require additional configuration to be used from Kotlin, and this document calls them
+  "third-party libraries".
+
+These two kinds of native libraries have different compatibility properties.
+
+## Platform libraries
+
+Platform libraries are shipped with the Kotlin/Native compiler.
+So, using different versions of Kotlin in the project results in getting different versions of platform libraries.
+For Apple targets (like iOS), platform libraries are generated based on an Xcode version this version of the compiler
+supports.
+
+Native APIs of libraries shipped with Xcode SDKs change with an Xcode version.
+Even when such changes are source- and binary-compatible within the native language,
+they might become breaking for Kotlin due to the way the interoperability is implemented.
+
+As a result, updating the Kotlin version in a project might bring a breaking change in platform libraries.
+This might matter in two cases:
+- There is a source breaking change in platform libraries that affects the compilation of the source code in your
+  project.
+  Usually it should be trivial to fix.
+- There is a binary breaking change in platform libraries that affects some of your dependencies.
```

Comment:
I believe these will now always manifest as partial linkage warnings/errors, right? Do we want to be more specific here and say that these issues will look like:
```
warning: <library> @ /code.kt:2:9: Function 'bar' can not be called: No function found for symbol 'foo/bar|bar(){}[100]'
warning: <missing declarations>: No function found for symbol 'foo/bar|bar(){}[100]'
```
and will produce a runtime exception?

we could also recommend promoting the warning to an error with the following:
```kotlin
kotlin {
    iosSimulatorArm64 {
        binaries.configureEach {
            freeCompilerArgs += "-Xpartial-linkage-loglevel=error"
        }
    }
}
```
if the user wants their linkage to be strict about these breakages